src/win32/tip/build_tip_forwarder_dll.py

```diff
@@ -0,0 +1,354 @@
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""This script builds am Arm64X pure forwarder DLL for Mozc TIPs on Windows.
+
+See the following document for more details:
+https://learn.microsoft.com/en-us/windows/arm/arm64x-build#building-an-arm64x-pure-forwarder-dll
+"""
+
+
+import argparse
+import dataclasses
+import os
+import pathlib
+import shutil
+import subprocess
+import tempfile
+from typing import Union
+
+from build_tools import mozc_version
+from build_tools import vs_util
+
+
+@dataclasses.dataclass
+class ForwarderInfo:
+  """Information about the forwarder DLL to be generated."""
+
+  branding: str
+  version: mozc_version.MozcVersion
+
+  def get_version_string(self, separator='.') -> str:
+    components = ['@MAJOR@', '@MINOR@', '@BUILD@', '@REVISION@']
+    return self.version.GetVersionInFormat(separator.join(components))
+
+  @property
+  def forwarder_dll_name(self) -> str:
+    return {
+        'Mozc': 'mozc_tip64x.dll',
+        'GoogleJapaneseInput': 'GoogleIMEJaTIP64X.dll',
+    }[self.branding]
+
+  @property
+  def x64_impl_dll_name(self) -> str:
+    return {
+        'Mozc': 'mozc_tip64.dll',
+        'GoogleJapaneseInput': 'GoogleIMEJaTIP64.dll',
+    }[self.branding]
+
+  @property
+  def arm64_impl_dll_name(self) -> str:
+    return {
+        'Mozc': 'mozc_tip64arm.dll',
+        'GoogleJapaneseInput': 'GoogleIMEJaTIP64Arm.dll',
+    }[self.branding]
+
+  @property
+  def file_description(self) -> str:
+    return {
+        'Mozc': 'Mozc TIP Module Forwarder',
+        'GoogleJapaneseInput': 'Google 日本語入力 TIP モジュール フォワーダー',
+    }[self.branding]
+
+  @property
+  def product_name(self) -> str:
+    return {
+        'Mozc': 'Mozc',
+        'GoogleJapaneseInput': 'Google 日本語入力',
+    }[self.branding]
+
+  @staticmethod
+  def _get_def_file_content(filename: str) -> str:
+    stem = str(pathlib.Path(filename).stem)
+    return '\n'.join(
+        [
+            'EXPORTS',
+            f'    DllGetClassObject = {stem}.DllGetClassObject',
+            f'    DllCanUnloadNow   = {stem}.DllCanUnloadNow',
+        ]
+    )
+
+  def get_def_file_content_for_x64(self) -> str:
+    return self._get_def_file_content(self.x64_impl_dll_name)
+
+  def get_def_file_content_for_arm64(self) -> str:
+    return self._get_def_file_content(self.arm64_impl_dll_name)
+
+  def get_rc_file_content(self) -> str:
+    comma_separated_versoin = self.get_version_string(',')
+    dot_separated_versoin = self.get_version_string('.')
+    file_description = self.file_description
+
+    original_file_name = None
```

Comment:
This is just overwritten by the next line.

Reply:
Removed.